BitFaster.Caching/Lru/ConcurrentLruBuilder.cs

```diff
@@ -35,13 +35,32 @@ public ConcurrentLruBuilder()
         internal ConcurrentLruBuilder(LruInfo<K> info)
             : base(info)
         {
+        }
+
+        /// <summary>
+        /// Evict after a variable duration specified by an IExpiry instance.
+        /// </summary>
+        /// <param name="expiry">The expiry that determines item time to live.</param>
+        /// <returns>A ConcurrentLruBuilder</returns>
+        public ConcurrentLruBuilder<K, V> WithExpiry(IExpiryCalculator<K, V> expiry)
+        {

```

Comment:
Should this be just an action delegate to start out:

`WithExpiry((k, v) => TimeSpan)`

Reply:
Only interface is public, the delegate version is now only part of unit tests.